### PR TITLE
ci: go mod tidy

### PR DESCRIPTION
**Pull request by ksubrmnn**
Creates a Github Action for go mod tidy to keep dependencies cleaned up

Dependent on #1768 

Please mark with X for applicable areas.

- New Functionality      [ ]
- Documentation          [ ]
- Install                [ ]
- Control Plane          [ ]
- CLI Tool               [ ]
- Certificate Management [ ]
- Networking             [ ]
- Metrics                [ ]
- SMI Policy             [ ]
- Security               [ ]
- Tests / CI System      [X]
- Other                  [ ]

Please answer the following questions with yes/no.

- Does this change contain code from or inspired by another project? If so, did you notify the maintainers and provide attribution?

No

### CODE REVIEW

**Comment by nojnhuh** [comment]
At the expense of yet another bash script, it would be great to invoke a check like this on the pre-push hook as well.

**Review by draychev — APPROVED** [review]

File: scripts/go-mod-tidy.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+go mod tidy
+if ! git diff --exit-code go.mod go.sum ; then
+    echo -e "\nPlease run 'go mod tidy' to clean up dependencies"
+    exit 1
+fi
```
Comment (shashankram) [inline scripts/go-mod-tidy.sh:7]: newline please.

Comment (ksubrmnn) [reply scripts/go-mod-tidy.sh]: Done

File: scripts/go-mod-tidy.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+go mod tidy
+if ! git diff --exit-code go.mod go.sum ; then
+    echo -e "\nPlease run 'go mod tidy' to clean up dependencies"
```
Comment (shashankram) [inline scripts/go-mod-tidy.sh:5]: This script is already running `go mod tidy`, so is there a reason why it should be run again as echoed in L5?

Comment (ksubrmnn) [reply scripts/go-mod-tidy.sh:5]: This script is also used by the CI so this is the message to let users know how to pass this check.

**Review by shashankram — APPROVED** [review]

**Review by shashankram — APPROVED** [review]

**Review by nojnhuh — APPROVED** [review]

**Pull request merged**